### PR TITLE
Switch RK methods to use an inf norm for `eigen_est`

### PR DESCRIPTION
**Pull request by oscardssmith**
fixes https://github.com/SciML/OrdinaryDiffEq.jl/issues/2198.

Originally, something seemed to be wrong since
```
function exrober(du, u, p, t)
    y₁, y₂, y₃ = u
    k₁, k₂, k₃ = p
    du .= vcat([-k₁ * y₁ + k₃ * y₂ * y₃,
      k₁ * y₁ - k₃ * y₂ * y₃ - k₂ * y₂^2,
      k₂ * y₂^2, ], u[4:end])
end
n = 100
jac_prototype = sparse(I(n+3))
jac_prototype[1:3, 1:3] .= 1.0
prob_ex_rober = ODEProblem(ODEFunction(exrober; jac_prototype), vcat([1.0,0.0,0.0], ones(n)),(0.0,100.0),(0.04,3e7,1e4))

julia> solve(prob_ex_rober, AutoTsit5(FBDF())).stats
SciMLBase.DEStats
Number of function 1 evaluations:                  4553
Number of function 2 evaluations:                  0
Number of W matrix evaluations:                    842
Number of linear solves:                           3278
Number of Jacobians created:                       3
Number of nonlinear solver iterations:             3258
Number of nonlinear solver convergence failures:   4
Number of fixed-point solver iterations:                     0
Number of fixed-point solver convergence failures:           0
Number of rootfind condition calls:                0
Number of accepted steps:                          859
Number of rejected steps:                          277
Maximum eigenvalue recorded:                       3623401289626744889992244603839849986785280

julia> solve(prob_ex_rober, AutoTsit5(FBDF(;autodiff=false))).stats
SciMLBase.DEStats
Number of function 1 evaluations:                  767728
Number of function 2 evaluations:                  0
Number of W matrix evaluations:                    179601
Number of linear solves:                           597222
Number of Jacobians created:                       33647
Number of nonlinear solver iterations:             498502
Number of nonlinear solver convergence failures:   111620
Number of fixed-point solver iterations:                     0
Number of fixed-point solver convergence failures:           0
Number of rootfind condition calls:                0
Number of accepted steps:                          34616
Number of rejected steps:                          240
Maximum eigenvalue recorded:                       3623401289626744889992244603839849986785280
```

but further investigation shows that this is expected given the magnitude of u.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Rebase?

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by oscardssmith** [comment]
rebased. This isn't quite ready to merge yet due to the issues with `autodiff=false` on AutoSwitch solvers for this problem (I suspect that the jacobian isn't getting refreshed correctly for some reason, but haven't figure out why yet).

File: src/composite_algs.jl [hunk 1]
```diff
@@ -15,7 +15,7 @@ function is_stiff(integrator, alg, ntol, stol, is_stiffalg)
     stiffness = abs(eigen_est * dt / alg_stability_size(alg)) # `abs` here is just for safety
     tol = is_stiffalg ? stol : ntol
     os = oneunit(stiffness)
-    bool = stiffness > os * tol
+    bool = !(stiffness <= os * tol)
```
Comment (ChrisRackauckas) [inline src/composite_algs.jl:18]: why this change?

**Review by ChrisRackauckas — APPROVED** [review]

File: src/composite_algs.jl (hunk 1 above)
Comment (oscardssmith) [reply src/composite_algs.jl:18]: in case `eigen_est` is `NaN`. It's a pretty good sign that things are going wrong, but since it's an estimate, if the estimate is `NaN` we really shouldn't be using an explicit solver.
